Sources/ProjectDescription/HelpersLocation.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+
+/// The location to a directory containing sources for `ProjectDescriptionHelpers`.
+public struct HelpersLocation: Codable, Equatable {
+
+    /// The type of location `local` or `git`.
+    public let type: LocationType
+
+    /// A `Path` to a directory containing `ProjectDescriptionHelpers`.
+    ///
+    /// Example:
+    /// ```
+    /// .local(path: "/User/local/bin")
+    /// ```
+    public static func local(path: Path) -> Self {
+        HelpersLocation(type: .local(path: path))
+    }
+
+    /// A `URL` to a `git` repository pointing at a `branch`.
+    ///
+    /// Example:
+    /// ```
+    /// .git(url: "https://git/helpers.git", branch: "main")
+    /// ```
+    public static func git(url: String, branch: String) -> Self {
+        HelpersLocation(type: .gitWithBranch(url: url, branch: branch))
+    }
+
+    /// A `URL` to a `git` repository pointing at a `tag`.
+    ///
+    /// Example:
+    /// ```
+    /// .git(url: "https://git/helpers.git", tag: "1.0.0")
+    /// ```
+    public static func git(url: String, tag: String) -> Self {
+        HelpersLocation(type: .gitWithTag(url: url, tag: tag))
+    }
+
+    /// A `URL` to a `git` repository pointing at a commit `hash`.
+    ///
+    /// Example:
+    /// ```
+    /// .git(url: "https://git/helpers.git", hash: "d06b4b3d")
+    /// ```
+    public static func git(url: String, hash: String) -> Self {
+        HelpersLocation(type: .gitWithHash(url: url, hash: hash))
+    }
+}
+
+// MARK: - Codable
+
+extension HelpersLocation {
+    public enum LocationType: Codable, Equatable {
+        case local(path: Path)
+        case gitWithBranch(url: String, branch: String)
+        case gitWithTag(url: String, tag: String)
+        case gitWithHash(url: String, hash: String)
+
+        enum CodingKeys: CodingKey  {
+            case local
+            case gitWithBranch
+            case gitWithTag
+            case gitWithHash
+        }
+
+        public func encode(to encoder: Encoder) throws {
```

Comment:
I'd test the custom en/decoding logic with `XCTAssertCodable` and create `HelpersLocationTests` for this